src/emm/clients/git_service.py

```diff
@@ -87,7 +87,10 @@ def fetch(self, directory: Optional[Path] = None) -> None:
             self.git[args]()
 
     def checkout(
-        self, revision: str, directory: Optional[Path] = None, branch_name: Optional[str] = None
+        self,
```

Comment:
I moved this files and all the files that touch external services (git, github, evergreen, etc) into a clients folder so that it is obvious they are wrappers over external services and we should avoid putting business logic in them.